test/neo/client/Client.d

```diff
@@ -222,6 +248,95 @@ public class Client
 
             return succeeded;
         }
+
+        /***********************************************************************
+
+            GetAll iterator "fruct" (foreach struct).
+
+        ***********************************************************************/
+
+        private struct GetAllFruct
+        {
+            /// Neo client instance to use to assign the request.
+            private Neo neo;
+
+            /// User's notifier delegate.
+            private Neo.GetAll.Notifier notifier;
+
+
+            /*******************************************************************
+
+                foreach iteration over the records returned by a GetAll request.
+                The calling Task is blocked until the request finishes.
+
+                Note: in this simple example, the case where the caller breaks
+                the foreach loop is not handled. In a real request
+                implementation, this should cleanly abort the request.
+
+            *******************************************************************/
+
+            public int opApply (
+                int delegate ( ref hash_t key, ref Const!(void)[] value ) dg )
+            {
+                int res;
+
+                auto task = Task.getThis();
+                assert(task !is null);
+
+                bool rq_finished;
+                void internalNotifier ( Neo.GetAll.Notification info, Neo.GetAll.Args args )
+                {
+                    this.notifier(info, args);
+
+                    with ( info.Active ) switch ( info.active )
+                    {
+                        case record:
+                            res = dg(info.record.key, info.record.value);
+
+                            // This simple iterator implementation does not
+                            // support ending the request via breaking the
+                            // foreach loop.
+                            assert(!res);
```

Comment:
Yes, it is. It just looks a bit funny to me: `break;` seriously breaks something.